Add no-GIL support

clin1234 opened the pull request.

clin1234 committed:
Disable building for limited API with no-GIL interpreters

clin1234 committed:
Update setup.py

clin1234 committed:
Fix minor oops

clin1234 committed:
FIx minor oops

clin1234 committed:
Another try

clin1234 committed:
Delete setup.cfg

clin1234 committed:
Call PyUnstable_Module_SetGIL

clin1234 committed:
Do not build extension.c with limited API

clin1234 committed:
Extremely silly spelling fix

oleavr commented on frida/_frida/extension.c:
```diff
@@ -6018,6 +6018,10 @@ PyInit__frida (void)
   module = PyModule_Create (&PyFrida_moduledef);
 
+#ifdef Py_GIL_DISABLED
+  PyUnstable_Module_SetGIL(module, Py_MOD_GIL_NOT_USED);
```
```suggestion
  PyUnstable_Module_SetGIL (module, Py_MOD_GIL_NOT_USED);
```

oleavr commented on frida/_frida/meson.build:
```diff
@@ -12,7 +12,6 @@ elif host_os_family != 'windows'
 endif
 
 extension = python.extension_module('_frida', 'extension.c',
-  limited_api: '3.7',
```
This seems like a regression, unless I'm missing something?

oleavr reviewed (changes requested):
Yay! 🙌

clin1234 committed:
Update frida/_frida/extension.c

Co-authored-by: Ole André Vadla Ravnås <oleavr@gmail.com>

clin1234 committed:
Restore building extension with limited API for now

Hopefully with https://peps.python.org/pep-0803/, no-GIL wheels will be stable (and meson supports this as well)
